R/tbl_split.R

```diff
@@ -1,66 +1,257 @@
-#' Split gtsummary table
+#' Split gtsummary table by rows and/or columns
 #'
 #' `r lifecycle::badge('experimental')`\cr
-#' The `tbl_split` function splits a single gtsummary table into multiple tables.
+#' The `tbl_split_by_rows` and `tbl_split_by_columns`` functions split a single
+#' gtsummary table into multiple tables.
+#' Both horizontal (column-wise, (that is, splits by columns in
+#' `x$table_body`) and vertical (row-wise) splits are possible.
 #' Updates to the print method are expected.
 #'
-#' @param x (`gtsummary`)\cr
-#'   gtsummary table
+#' @param x (`gtsummary` or `list`)\cr
+#'   gtsummary table.
 #' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
 #'   variables at which to split the gtsummary table rows (tables
-#'   will be separated after each of these variables)
+#'   will be separated after each of these variables).
+#' @param keys ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   columns to be repeated in each table split. It defaults to `first_column(x)`
+#'   if missing (usually label column).
+#' @param groups (list of `character` vectors)\cr
+#'   list of column names that appear in `x$table_body`.
+#'   Each group of column names represent a different table in the output list.
+#' @param footnotes (`string`)\cr
+#'   can be either `"first"`, `"each"`, or `"last"`, to locate global footnotes
+#'   only on the first, in each, or in the last table, respectively. It defaults
+#'   to `"each"`. Reference footnotes are always present wherever they appear.
+#' @param caption (`string`)\cr
+#'   similarly to footnotes, makes the titles, and subtitles appear only on the
+#'   `"first"` or `"each"` table. It defaults to `"each"`.
 #' @inheritParams rlang::args_dots_empty
 #'
-#' @return `tbl_split` object
-#' @name tbl_split
+#' @return `tbl_split` object. They can also be identified further by subclass
+#'   definitions `tbl_split_by_rows` and `tbl_split_by_columns`.
+#'
+#' @details
+#' Run [show_header_names()] to print all column names to split by.
 #'
 #' @examples
-#' tbl <-
-#'   tbl_summary(trial) |>
-#'   tbl_split(variables = c(marker, grade))
+#' # create standard table
```

Comment:
Can you structure this like the other Examples sections, where we number the examples and print the results?

Also, let's make the table smaller, because when these results are printed on the pkgdown site they will be soooo long.

Reply:
good point